apps/condo/domains/user/schema/CheckUserExistenceService.js

```diff
@@ -0,0 +1,133 @@
+        code: BAD_USER_INPUT,
+        type: 'INVALID_PHONE_NUMBER',
+        message: 'Invalid phone number',
+    },
+    TOKEN_NOT_FOUND: {
+        query: 'checkUserExistence',
+        variable: ['data', 'token'],
+        code: NOT_FOUND,
+        type: 'TOKEN_NOT_FOUND',
+        message: 'Token not found',
+    },
+    DV_VERSION_MISMATCH: {
+        query: 'checkUserExistence',
+        variable: ['data', 'dv'],
+        code: BAD_USER_INPUT,
+        type: DV_VERSION_MISMATCH,
+        message: 'Wrong value for data version number',
+    },
+    WRONG_SENDER_FORMAT: {
+        query: 'checkUserExistence',
+        variable: ['data', 'sender'],
+        code: BAD_USER_INPUT,
+        type: WRONG_FORMAT,
+        message: 'Invalid format of "sender" field value',
+        correctExample: '{ dv: 1, fingerprint: \'example-fingerprint-alphanumeric-value\'}',
+    },
+}
+
+const CheckUserExistenceService = new GQLCustomSchema('CheckUserExistenceService', {
+    types: [
+        {
+            access: true,
+            type: 'input CheckUserExistenceInput { dv: Int!, sender: SenderFieldInput!, confirmActionToken: ID!, userType: UserTypeType! }',
+        },
+        {
+            access: true,
+            type: 'type CheckUserExistenceOutput { userExists: Boolean!, nameIsSet: Boolean!, emailIsSet: Boolean!, phoneIsSet: Boolean!, passwordIsSet: Boolean! }',
+        },
+    ],
+
+    queries: [
+        {
+            access: access.canCheckUserExistence,
+            schema: 'checkUserExistence(data: CheckUserExistenceInput!): CheckUserExistenceOutput',
+            doc: {
+                summary: 'Using an action token with a verified phone number,' +
+                    ' checks whether a user with the specified type (resident, staff, service) is registered.\n' +
+                    'As a result, information about the completion of some important fields (name, email, phone, password) is also returned.',
+                errors: ERRORS,
+            },
+            resolver: async (parent, args, context) => {
+                const { data } = args
+                const { confirmActionToken, userType } = data
+
+                await checkDayAnonymousRequestLimitCounters(context, 'checkUserExistence', context.req.ip)
+
+                checkDvAndSender(data, ERRORS.DV_VERSION_MISMATCH, ERRORS.WRONG_SENDER_FORMAT, context)
+
+                if (!confirmActionToken) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                const action = await ConfirmPhoneAction.getOne(context,
+                    {
+                        token: confirmActionToken,
+                        expiresAt_gte: new Date().toISOString(),
+                        completedAt: null,
+                        isPhoneVerified: true,
+                    },
+                )
+                if (!action) throw new GQLError(ERRORS.TOKEN_NOT_FOUND, context)
+
+                const phoneFromAction = normalizePhone(action.phone) || null
```

Comment:
Should we check it here? Action must validateInput by itself...

Reply:
The only case I can find it reasonable, if ConfirmAction will support landline, and mutations / not

Reply:
But its definitely not the case :)

Reply:
> Should we check it here? Action must validateInput by itself...

You are right, I will remove normalization from here :)